packages/backend/src/apis/quadlet-api-impl.ts

```diff
@@ -60,6 +62,8 @@ export class QuadletApiImpl extends QuadletApi {
     if (!quadlet.service)
```

Comment:
Suggestion: refactor with the start method as the multiple checks seem to be the same

Reply:
I don't want to refactor those check, as the error message contain `start` and `stop`, and I don't want to write some function taking as argument `operation: 'start' | 'stop'` :/

Reply:
Adding a try catch in the start / stop, to concatenate the error, is a risky move, let's say I catch an error, and concatenate `err.message` => I can't do that, err would be of type unknown, so I need to a use `isError`, adding an if clause inside the catch, to use the message.

I think factorising those few lines is not worth the effort :/ this keep the readability pretty simple

Reply:
you would need to update the checkQuadlet signature

```ts
private checkQuadlet(quadlet: Quadlet): quadlet is (Quadlet & { service: string, isTemplate: false }) {
 // ...
}
```

this method could be externalised and reuse in other places, as you mention in other comment

Reply:
but as mentionned in https://github.com/podman-desktop/extension-podman-quadlet/pull/752#discussion_r2215554675 this add extra complexity, and for two lines duplicated, already covered by tests, IMO this is not worth the effort

Reply:
I opened a PR based on your branch here for my implem

https://github.com/axel7083/extension-podman-quadlet/pull/4/files